// @specular-aura/voice-modulation/src/components/modulations/ModulationMap.js
import alien1Transform from "./alien1";
import alien2Transform from "./alien2";
import alienRobot1Transform from "./alienRobot1";
import baneTransform from "./bane";
import chunkedReversedTransform from "./chunkedReversed";
import churchTransform from "./church";
import cymbalTransform from "./cymbal";
import deepSynthTransform from "./deepSynth";
import distopianPASpeakerTransform from "./distopianPASystem";
import evilRobotTransform from "./evilRobot";
import jumbledUpTransform from "./jumbledUp";
import megaphoneTransform from "./megaphone";
import mosquitoTransform from "./mosquito";
import oldRadioTransform from "./oldRadio";
import phaserTransform from "./phaser";
import pitchTransform from "./pitch";
import poorTelephoneTransmissionTransform from "./poorTelephoneTransmission";
import portableRadioTransform from "./portableRadio";
import radioTuningTransform from "./radioTuning";
import reverseReverbTransform from "./reverseReverb";
import ringModulatorTransform from "./ringModulator";
import robot1Transform from "./robot1";
import robot2Transform from "./robot2";
import robot3Transform from "./robot3";
import satanTransform from "./satan";
import sauronTransform from "./sauron";
import slowWobbleTransform from "./slowWobble";
import speedTransform from "./speed";
import synthTransform from "./synth";
import telephoneTransform from "./telephone";
import tinyRobotTransform from "./tinyRobot";
import trollTransform from "./troll";
import underwaterTransform from "./underwater";
import vibratoTransform from "./vibrato";

const modulationMaps = {
  distopianPASystem: distopianPASpeakerTransform,
  church: churchTransform,
  alien1: alien1Transform,
  alien2: alien2Transform,
  alienRobot1Transform,
  bane: baneTransform,
  chunkedReversed: chunkedReversedTransform,
  cymbal: cymbalTransform,
  deepSynth: deepSynthTransform,
  evilRobot: evilRobotTransform,
  jumbledUp: jumbledUpTransform,
  megaphone: megaphoneTransform,
  mosquito: mosquitoTransform,
  oldRadio: oldRadioTransform,
  phaser: phaserTransform,
  pitch: pitchTransform,
  poorTelephoneTransmission: poorTelephoneTransmissionTransform,
  portableRadio: portableRadioTransform,
  radioTuning: radioTuningTransform,
  reverseReverb: reverseReverbTransform,
  ringModulator: ringModulatorTransform,
  robot1: robot1Transform,
  robot2: robot2Transform,
  robot3: robot3Transform,
  satan: satanTransform,
  sauron: sauronTransform,
  slowWobble: slowWobbleTransform,
  speed: speedTransform,
  synth: synthTransform,
  telephone: telephoneTransform,
  tinyRobot: tinyRobotTransform,
  troll: trollTransform,
  underwater: underwaterTransform,
  vibrato: vibratoTransform,
};

export default modulationMaps;
