// @specular-aura/voice-modulation/src/components/VoiceModulation.js
// import React from "react";
import audioBufferToWaveBlob from "./helpers/audioBufferToWaveBlob";
import modulationMaps from "./modulations/ModulationMap";
const VoiceModulation = () => {
  return <div>Lalalalalallalalalalla</div>;
};

const useVoiceModulation = (currentSong, setCurrentSong, transformName) => {
  const transformFunc = modulationMaps[transformName];
  const changeVoice = async () => {
    console.log(currentSong.url);
    if (currentSong) {
      const arrayBuffer = await (await fetch(currentSong?.url)).arrayBuffer();
      let ctx = new AudioContext();
      const audioBuffer = await ctx.decodeAudioData(arrayBuffer);
      let outputAudioBuffer = await transformFunc(audioBuffer);
      let outputWavBlob = await audioBufferToWaveBlob(outputAudioBuffer);
      let audioUrl = window.URL.createObjectURL(outputWavBlob);
      console.log(currentSong.url);
      setCurrentSong({
        id: Math.floor(Math.random() * 100),
        title: `${currentSong.title} Modified`,
        description: "Original Soundtrack",
        artist: currentSong.artist,
        image: currentSong.image,
        type: currentSong.type,
        url: audioUrl,
      });
    }
  };
  return changeVoice;
};

export default VoiceModulation;
export { useVoiceModulation };
